RevenueCatUI/Views/RemoteImage.swift

```diff
@@ -288,27 +250,14 @@ private struct ColorSchemeRemoteImage<Content: View>: View {
             }
             #endif
 
-            // 1. Only attempt to fetch the low res again if we don't have it
-            if self.lowResCachedImage == nil, let url = self.lowResURLForScheme {
-                self.lowResCachedImage = self.fileRepository
-                    .getCachedFileURL(for: url, withChecksum: nil)?.asImageAndSize
-                self.imageLoadedFrom = .lowResFileRepository
+            guard self.highResFileLoader.result == nil else {
+                return
             }
 
-            // 2. Fetch the high res to replace the initial low res
-            do {
-                let highResCachedImage = try await self.fileRepository.generateOrGetCachedFileURL(
-                    for: self.highResFileUrlForScheme,
-                    withChecksum: nil
-                ).asImageAndSize
-                self.highResCachedImage = highResCachedImage
-                self.imageLoadedFrom = .highResFileRepository
-            } catch {
+            await self.lowResFileLoader.load()
+            await self.highResFileLoader.load()
```

Comment:
We can expedite this a smidge by doing asynchronous let. This will matter more if the detached task is not used anymore. Something like this:

```suggestion
            async let low: Void = await self.lowResFileLoader.load()
            async let high: Void = await self.highResFileLoader.load()
            await (low, high)
```

Reply:
Woah!!